packages/twenty-front/src/modules/context-store/hooks/useContextStoreSelectedRecords.ts

```diff
@@ -0,0 +1,109 @@
+import { contextStoreCurrentObjectMetadataIdState } from '@/context-store/states/contextStoreCurrentObjectMetadataIdState';
+import { contextStoreTargetedRecordsFiltersState } from '@/context-store/states/contextStoreTargetedRecordsFilters';
+import { contextStoreTargetedRecordsState } from '@/context-store/states/contextStoreTargetedRecordsState';
+import { useObjectMetadataItemById } from '@/object-metadata/hooks/useObjectMetadataItemById';
+import { RecordGqlFields } from '@/object-record/graphql/types/RecordGqlFields';
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { turnFiltersIntoQueryFilter } from '@/object-record/record-filter/utils/turnFiltersIntoQueryFilter';
+import { useFindManyParams } from '@/object-record/record-index/hooks/useLoadRecordIndexTable';
+import { makeAndFilterVariables } from '@/object-record/utils/makeAndFilterVariables';
+import { useRecoilValue } from 'recoil';
+
+const getFilterForSelectedRecords = (
+  selectedRecordIds: string[] | 'all',
+  excludedRecordIds: string[],
+  queryFilter: any,
+) => {
+  if (selectedRecordIds === 'all') {
+    if (excludedRecordIds.length > 0) {
+      return makeAndFilterVariables([
+        queryFilter,
+        {
+          not: {
+            id: {
+              in: excludedRecordIds,
+            },
+          },
+        },
+      ]);
+    }
+    return queryFilter;
+  }
+
+  return makeAndFilterVariables([
+    queryFilter,
+    {
+      id: {
+        in: selectedRecordIds,
+      },
+    },
+  ]);
+};
+
+export const useContextStoreSelectedRecords = ({
+  limit = undefined,
+  recordGqlFields,
+}: {
+  limit?: number;
+  recordGqlFields?: RecordGqlFields;
+}) => {
+  const contextStoreTargetedRecords = useRecoilValue(
+    contextStoreTargetedRecordsState,
+  );
+
+  const contextStoreCurrentObjectMetadataId = useRecoilValue(
+    contextStoreCurrentObjectMetadataIdState,
+  );
+
+  const contextStoreTargetedRecordsFilters = useRecoilValue(
+    contextStoreTargetedRecordsFiltersState,
+  );
+
+  const { objectMetadataItem } = useObjectMetadataItemById({
+    objectId: contextStoreCurrentObjectMetadataId,
+  });
+  const queryFilter = turnFiltersIntoQueryFilter(
+    contextStoreTargetedRecordsFilters,
+    objectMetadataItem?.fields ?? [],
+  );
+
+  const { selectedRecordIds, excludedRecordIds } = contextStoreTargetedRecords;
+
+  // Determine if we should skip the query based on the following conditions:
+  // 1. We're not selecting all records (selectedRecordIds !== 'all')
+  // 2. Either:
+  //    a) No specific records are selected (selectedRecordIds.length === 0)
+  //    b) We're only requesting the 'id' field (which we already have)
+  const isOnlyRequestingId =
```

Comment:
nit: Usually extracting to a function with a good name can be explicit enough and you won't need a comment. Maybe try extracting isOnlyRequestingId and skip logic in a function
```typescript
  const shouldSkipQuery = (): boolean => {
    const isOnlyRequestingId =
      Object.keys(recordGqlFields ?? {}).length === 1 &&
      recordGqlFields?.id === true;

    return (
      selectedRecordIds !== 'all' &&
      (selectedRecordIds.length === 0 || isOnlyRequestingId)
    );
  };
  
const skip = shouldSkipQuery();
  ````